test-data/unit/check-typeddict.test

```diff
@@ -2087,50 +2083,6 @@ def f(arg: TD) -> None:
 [builtins fixtures/dict.pyi]
 [typing fixtures/typing-typeddict.pyi]
 
-[case testOperatorContainsNarrowsTypedDicts_final]
-# flags: --warn-unreachable
-from __future__ import annotations
-from typing import assert_type, TypedDict, Union
-from typing_extensions import final
-
-@final
-class DFinal(TypedDict):
-    foo: int
-
-
-class DNotFinal(TypedDict):
-    bar: int
-
-
-d_not_final: DNotFinal
-
-if 'bar' in d_not_final:
-    assert_type(d_not_final, DNotFinal)
-else:
-    spam = 'ham'  # E: Statement is unreachable
-
-if 'spam' in d_not_final:
-    assert_type(d_not_final, DNotFinal)
-else:
-    assert_type(d_not_final, DNotFinal)
-
-d_final: DFinal
-
-if 'spam' in d_final:
-    spam = 'ham'  # E: Statement is unreachable
```

Comment:
Is this test really useless? E.g. this seems useful and I'm not sure where else it is tested

Reply:
Good point, brought it back in a68b87abff4f38d0ebf469e986f1aad98cbe0de5 (and also removed more `@final`s).